regional_mom6/regional_mom6.py

```diff
@@ -1222,6 +1225,204 @@ def FRE_tools(self, layout):
         )
         self.layout = layout
 
+    def setup_run_directory(self, rmom6_path, surface_forcing="era5", using_payu=False):
+        """Sets up the run directory for MOM6. Creates a symbolic link
+        to the input directory, and creates a payu configuration file
+        if payu is being used.
+
+        Args:
+            rmom6_path [str]:   The path to where the regional_mom6 package is installed. This is needed to find the default run directory that this function builds on
+            surface_forcing (Optional[str]): The surface forcing to use. One of ``era5`` or ``jra``.
+            using_payu (Optional[bool]): Whether or not to use payu to run the model. If True, a payu configuration file will be created.
+
+        """
+
+        ## Copy the default directory to the run directory
+
+        subprocess.run(
+            f"cp {str(Path(rmom6_path) / 'regional_mom6' / 'default_rundir' / surface_forcing)}_surface/* {str(self.mom_run_dir)}",
+            shell=True,
+        )
+        ## Make symlinks between run and input directories
+        if not (self.mom_run_dir / "inputdir").exists():
+            os.symlink(str(self.mom_input_dir), str(self.mom_run_dir / "inputdir"))
+        if not (self.mom_input_dir / "rundir").exists():
+            os.symlink(str(self.mom_run_dir), str(self.mom_input_dir / "rundir"))
+
+        ## Get mask table information
+        ncpus = 10
+        mask_table = None
+        for i in os.listdir(f"{self.mom_input_dir}"):
+            if "mask_table" in i:
+                mask_table = i
+                a = mask_table.split(".")[1]
+                b = mask_table.split(".")[2].split("x")
+                ncpus = int(b[0]) * int(b[1]) - int(a)
```

Comment:
An alternative could be to parse from the mask file directly (the filename could lie!)

```python
mask_tables = self.mom_input_dir.glob("mask_table.*")
for mask_table in mask_tables:
  with open(mask_table, "r") as f:
    masked_cpus = int(f.readline())
    x, y = (int(v) for v in f.readline().split(", "))
```

Otherwise, at least:
```suggestion
        for p in self.mom_input_dir.glob("mask_table.*"):
          mask_table, masked, layout = p.name.split(".")
          x, y = (int(v) for v in layout.split("x"))
          ncpus = (x * y) - masked
```

But, what if there are multiple mask tables? This (both mine and yours) will default to the lexicographically last. Also, the layout we're going to use is set by `FRE_tools()`, but the core count is determined here, at run time. If a mask file does exist, but `FRE_tools()` wasn't called, `self.layout` won't exist and you'll hit an error later on. If a different mask file exists to the one `FRE_tools()` creates, the core count and layout could be out of sync.

Reply:
Yeah I'm not sure how to handle all the possibilities here. The point of this function is to really hold the hand of a new user. I'll get it to print a warning if there are multiple mask tables to prompt them to delete or move unwanted ones